### ISSUE TITLE
Lag on input (severe with a held key)

### ISSUE THREAD

**Issue by patrick-kidger** [zellij-org/zellij#1556]
I recently discovered Zellij and it seems pretty cool!

However, I've come across poor performance in one particular case. I'm connecting to a remote Linux machine over SSH; Zellij is installed on the remote machine; the local machine is using [Chrome Secure Shell](https://chrome.google.com/webstore/detail/secure-shell/iodihamcpbpeioajjeobimgagajmlibd).

In this case, I'm getting pretty bad input lag. After typing a single key it only appears after about (eyeball'd guesstimate) 200ms. When holding down a key, the stream of inputs seems to back up. For example I might hold the enter key (on the terminal) for 5 seconds, release it, and then have the window continue scrolling for another 5 seconds before stopping. This behaviour is common across programs (fish, nvim, etc.)

What's pretty weird is that I'm only getting this issue in this one scenario. Using Zellij with a different terminal is fine. (Which is why I mention it above.) Using Zellij locally is fine. The SSH connection is also fine: using no multiplexer, or tmux, has no lag at all.

**Info**

Unfortunately I can't attach any output from a corporate machine. :/

**Further information**

This is Zellij version 0.30.0, installed via `cargo install zellij`.

I appreciate this report may be borderline-unreproducible on your end! If you've seen anything like this before I'd be happy to try a few possible fixes and see if anything sticks. Other than that, I just thought I'd make you aware of this issue.

**Comment by raphCode** [comment]
Hi, thanks for reaching out!
You could try to get some CPU / system load to see if some processes involved (zellij server, zellij client, ssh, chrome native shell) show something suspicious.
How does tmux compare in performance?

**Comment by dzerus3** [comment]
I have the same issue. I can reproduce it when:
1. Using SSH under a headless [crouton](https://github.com/dnschneid/crouton) in the Chrome Shell.
2. Using the above-mentioned Chrome Secure Shell extension.
3. Using SSH from [crostini](https://chromeos.dev/en/linux).

It is noteworthy that I can *not* reproduce this issue inside that same crouton when it uses its own graphical environment. Therefore, the issue is somehow specific either to Chrome Shell or ChromeOS.

I don't have the time to run the profiler right now @raphCode, but I'll try to do it within the next day or two. I also want to try to test out whether running Zellij locally will result in the same issues, and I'll try to attach a GIF to demonstrate the issue if I can.

**Comment by dzerus3** [comment]
The slowdown is there when running Zellij locally as well as remotely, though it is less noticeable. Therefore the issue is not with SSH specifically, but with crosh.

I haven't yet tested whether this bug exists when running the extension in Chrome not on Chrome OS and have not run the profiler, but I have recorded a video which demonstrates the issue.

[screen-capture.webm](https://user-images.githubusercontent.com/60197566/180353059-6c3e05a0-beff-4741-84dc-778355f008d6.webm)

Please, ignore my fortunes.

**Comment by raphCode** [comment]
Interesting information, I will try to investigate.

**Comment by dzerus3** [comment]
Original plan was to run the benchmark "within the next day or two," but I cannot successfully get it to compile on my device. Afraid I have to retract that promise for now, sorry.

**Comment by a-kenji** [comment]
@dzerus3,
If you are using cargo install, did you try installing it with the information in the readme?

`cargo install --locked zellij`

**Comment by dzerus3** [comment]
> @dzerus3,
> 
> If you are using cargo install, did you try installing it with the information in the readme?

Yeah, the issue is that whatever repositories my device uses do not include a recent version of Cargo, and it fails to compile because of that every time I try.

I would like to try again later, but as it is now I use it only for SSH and it's not the most urgent of issues for me.

**Comment by dzerus3** [comment]
> I have yet to test it using a version Chromium on a different device though. I'll try and do that for now.

Chromium extension for me fails to load because _Loading pnacl program..._. Judging from a another issue report here, I'm not the only one who's encountered that issue.

Sorry everyone, I think the powers that be have conspired to prevent me from testing this out any further 😁
